pkg/controller/stacks/templates/stackdefinition_controller_test.go

```diff
@@ -0,0 +1,239 @@
+    - crd: '*.servicenetworking.gcp.crossplane.io/v1alpha3'
+    - crd: '*.gcp.crossplane.io/v1alpha3'
+  license: Apache-2.0
+`
+)
+
+func TestStackDefinitionController(t *testing.T) {
+	cases := map[string]struct {
+		reason string
+		test   func(c client.Client, mgr manager.Manager) error
+	}{
+		"CreateStack": {
+			reason: "should create a stack which doesn't exist yet",
+			test: func(c client.Client, mgr manager.Manager) error {
+				namespace := "test-stackdefinition-controller"
+				ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+				defer cancel()
+
+				// Set up test namespace
+				n := &corev1.Namespace{}
+				n.SetName(namespace)
+				if err := c.Create(ctx, n); err != nil {
+					t.Log("CreateStack(): returning error result when trying to create namespace.", "err", err)
+					return err
+				}
+
+				// Add our test data to the "cluster"
+				u := &unstructured.Unstructured{}
+				if err := yaml.Unmarshal([]byte(testStackDefinition), u); err != nil {
+					t.Log("CreateStack(): returning error result when unmarshaling stack definition.", "err", err)
+					return err
+				}
+
+				u.SetNamespace(namespace)
+				if err := c.Create(ctx, u); err != nil {
+					t.Log("CreateStack(): returning error result when creating stack definition.", "err", err)
+					return err
+				}
+
+				name := u.GetName()
+				nn := types.NamespacedName{
+					Namespace: namespace,
+					Name:      name,
+				}
+
+				// A paranoid test may check for the precondition that the expected result Stack
+				// does not exist at this time. Let's be paranoid.
+				s := &v1alpha1.Stack{}
+				if err := c.Get(ctx, nn, s); err == nil {
+					t.Error("CreateStack(): returning error result when fetching created stack\ngot = err: nil\nwant = err: not found", "err", err, "stack", s)
+				} else if !kerrors.IsNotFound(err) {
+					t.Errorf("CreateStack(): returning error result when fetching created stack\ngot = err: %v\nwant = err: not found", err)
+				}
+
+				// Now we reconcile
+				reconciler := StackDefinitionReconciler{
+					Client: c,
+					Log:    ctrl.Log,
+				}
+				reconcileRequest := ctrl.Request{
+					NamespacedName: nn,
+				}
+				res, err := reconciler.Reconcile(reconcileRequest)
+
+				// We want our reconcile to succeed
+				if res.Requeue || res.RequeueAfter != 0 || err != nil {
+					return fmt.Errorf("CreateStack(): reconcile did not complete successfully.\ngot = result: %v, err: %v\nwant = Requeue: false, RequeueAfter: 0, err: nil", res, err)
+				}
+
+				// Now that our reconcile has succeeded, we expect the stack to exist
+				ss := &v1alpha1.Stack{}
+				if err := c.Get(ctx, nn, ss); err != nil {
+					t.Log("CreateStack(): returning error result when fetching created stack", "err", err, "stack", s)
+					return err
+				}
+
+				// NOTE the test does not check the fields of the stack. Ideally it would, for correctness guarantees
+
+				return nil
+			},
+		},
+	}
+
+	var cfg *rest.Config
```

Comment:
@suskin this is failing because you are passing a `nil` config to the framework, which will make it assume that you want to start a local control plane by loading `kube-apiserver` and `etcd` from `/usr/local/kubebuilder/bin`. This behavior is part of `envtest` (read [here](https://godoc.org/sigs.k8s.io/controller-runtime/pkg/envtest)) and something we could change in the `crossplane-runtime` wrapper (we could also just not wrap `envtest` at all but it has some useful setup of installing CRDs which we would effectively just be rewriting). If you have feature requests for how you would like this changed / expanded I would be happy to add them to my queue :)